src/Shared/PropertyActivator/PropertyActivator.cs

```diff
@@ -56,6 +56,32 @@ public object Activate(object instance, TContext context)
         ArgumentNullException.ThrowIfNull(activateAttributeType);
         ArgumentNullException.ThrowIfNull(createActivateInfo);
 
+        var properties = GetActivatableProperties(type, activateAttributeType, includeNonPublic);
+        return properties.Select(createActivateInfo).ToArray();
+    }
+
+    public static PropertyActivator<TContext>[] GetPropertiesToActivate<TAttribute>(
+        Type type,
+        Func<PropertyInfo, TAttribute, PropertyActivator<TContext>> createActivateInfo,
+        bool includeNonPublic)
+        where TAttribute : Attribute
+    {
+        ArgumentNullException.ThrowIfNull(type);
+        ArgumentNullException.ThrowIfNull(createActivateInfo);
+
+        var properties = GetActivatableProperties(type, typeof(TAttribute), includeNonPublic);
+        return properties.Select(property =>
+        {
+            var attribute = property.GetCustomAttribute<TAttribute>()!;
+            return createActivateInfo(property, attribute);
+        }).ToArray();
```

Comment:
This is interesting! Pranav did a cull of all our uses of `Select(lambda)` but looks like this - which I see you're bringing forwards from the old code - escaped, perhaps because it's in `Shared`.

Totally fine to leave as-is but if you decide to make other changes here might be nice to swap this over to building the array manually.

Reply:
We can file an issue to track removing it.